### ISSUE TITLE
Standardized naming convention for limited exposure variables in gdscript

### ISSUE THREAD

**Issue by LimestaX** [godotengine/godot-proposals#8868]
### Describe the project you are working on

A plugin within a team

### Describe the problem or limitation you are having in your project

Functions with limited in-script use are being consistently confused with function intended to be overridden, and team members are accessing functions and variables which can cause further confusion.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Proposal is a refined solution for #641 by creating a new naming convention standard for variables which are hidden only from the editor.
Adding hidden enforcement to the engine would increase complexity and make it less usable for users.
This proposal is designed to clean up workflows, optionally, through limited exposure in the editor, which can be disabled if needed

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

When a user is developing helper functions or variables that they do not wish to expose in the editor, for cleanliness and for team-member exposure, they can add a double underscore in-front of any variable or function.

Example use: 
![image](https://github.com/godotengine/godot-proposals/assets/45181474/e16f2178-6ce7-48fa-b931-677af02849a2)
![image](https://github.com/godotengine/godot-proposals/assets/45181474/01e86da8-11d4-47ca-9370-f6f3e2eef8a8)

Functions and variables are treated as any other in the engine, with full exposure to any built in function as normal, all name casing is still followed

The naming convention is enforced only in the editor level, proposing the following to be implemented to the editor:

1. A Project-level setting under `debug/gdscript/warnings/called_hidden_variable`
This includes the standard Ignore/**Warn**/Error enum, responding accordingly if you call/use a hidden function or variable, default warn
2.  A Project-level setting under `editor/script/expose_hidden_autofill` **false**
This allows the user to make hidden functions and variables appear in the autofill, by default is disabled
3. A Project-level setting under `editor/script/expose_hidden_signals` **false** _tentative naming_
This allows the user to make hidden signals appear in the Signal menu **_maybe we make create a hidden exposure icon to be implemented where needed somewhere in the editor? Maybe toggle icon's exposure editor wide?_**
4. Add toggles alongside current ones in various editor prompts, exposing hidden variables to prompts and menus
Includes, but not limited to: 
-signals' prompt menu
-animations
-animation trees

### If this enhancement will not be used often, can it be worked around with a few lines of script?

No, this would require modifying several parts of the editor code.

### Is there a reason why this should be core and not an add-on in the asset library?

This is adding to workflows, and would be also create a standard for implementing editor hidden functions, allowing for an easy to understand concept, current convention is non existent, and using a single underscore conflicts with other conventions for functions intended to be overridden

By default this does not change anyone's workflow, but can help with cleaning up workflow and auto-exposure of functions when making interconnected systems, and if we call them as exposure-limited or hidden instead of private, it will not be confused with C# convention

**Comment by Calinou** [comment]
I would prefer actual `@private` annotations over a naming convention, which is easier to enforce and provides autocompletion.

**Comment by JeckDev** [comment]
I've just been picking up GDScript this year, so take this for what it's worth, but I actually really like the idea of making the enforcement behavior configurable in the editor with a warn/error/etc option, and leaving all "private" members still technically accessible if need be.

Having all the internals accessible right now (no access control for members) has been very cool. I feel confident that if I need to I can meddle with other scripted objects any way I'd like. 

Hypothetically, if access control was enforced, there could be situations where I'd need to edit external scripts' source (say, an editor plugin) to modify access control for some new feature, and doing so on scripts maintained by someone else means having to deal with merge conflicts in the future.

tl;dr— I like marking things private (and also prefer a "@private" annotation) for warnings/autocompletion but don't want scripts to error out if I violate access control.
